spec/cyanobyte.schema.json

```diff
@@ -0,0 +1,216 @@
+          "type": "string"
+        },
+        "version": {
+          "description": "Version number for this document using kind of SymVer.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "title",
+        "description",
+        "contact",
+        "license",
+        "version"
+      ]
+    },
+    "i2c": {
+      "description": "I2C settings",
+      "type": "object",
+      "properties": {
+        "addressType": {
+          "description": "Addressing type of the component.",
+          "type": "string",
+          "enum": [
+            "7-bit",
+            "10-bit"
+          ]
+        },
+        "address": {
+          "description": "I2C Address of the component",
+          "type": "integer"
+        },
+        "addressMask": {
+          "description": "Address mask for addresses that can be altered"
+        }
+      },
+      "required": [
+        "addressType",
+        "address",
+        "addressMask"
+      ]
+    },
+    "registers": {
+      "description": "Chip registers",
+      "type": "array",
+      "items": {
+        "description": "Outer object",
+        "type": "object",
+        "patternProperties": {
+          "^.*$": {
+            "description": "Chip register",
+            "type": "object",
+            "properties": {
+              "title": {
+                "description": "Title of the register",
+                "type": "string"
+              },
+              "description": {
+                "description": "Description of the register",
+                "type": "string"
+              },
+              "address": {
+                "description": "Address of the register",
+                "type": "integer"
+              },
+              "length": {
+                "description": "Length of the register in bits.",
+                "type": "integer"
+              }
+            },
+            "required": [
+              "title",
+              "description",
+              "address",
+              "length"
+            ]
+          }
+        },
+        "additionalProperties": false
+      }
+    },
+    "functions": {
+      "description": "Chip functions",
+      "type": "array",
+      "items": {
+        "description": "Outer object",
+        "type": "object",
+        "patternProperties": {
+          "^.*$": {
+            "description": "Chip function",
+            "type": "object",
+            "properties": {
+              "title": {
+                "description": "Title of the function",
+                "type": "string"
+              },
+              "description": {
+                "description": "Description of the function",
+                "type": "string"
+              },
+              "register": {
+                "description": "Register of the function",
+                "type": "string"
+              },
+              "readWrite": {
```

Comment:
This field is missing in the example for the example's `critLock`

Reply:
I removed the `critLock` in the actual file I was testing as the example in the `cyanobyte.md` was unfinished.